uaclient/config.py

```diff
@@ -751,7 +751,104 @@ def _attached_status(self) -> Dict[str, Any]:
                 response["contract"]["tech_support_level"] = supportLevel
         return response
 
-    def status(self, show_beta=False) -> Dict[str, Any]:
+    def _get_entitlement_information(
+        self, entitlements: List[Dict[str, Any]], entitlement_name: str
+    ) -> Dict[str, Any]:
+        """Extract information from the entitlements array."""
+        for entitlement in entitlements:
+            if entitlement.get("type") == entitlement_name:
+                return {
+                    "entitled": "yes" if entitlement.get("entitled") else "no",
+                    "auto_enabled": "yes"
+                    if entitlement.get("obligations", {}).get(
+                        "enableByDefault"
+                    )
+                    else "no",
+                    "affordances": entitlement.get("affordances", {}),
+                }
+        return {"entitled": "no", "auto_enabled": "no", "affordances": {}}
+
+    def _simulate_status(self, token: str) -> Dict[str, Any]:
+        """Return a status dictionary based on a token."""
+        from uaclient.contract import (
+            get_available_resources,
+            get_contract_information,
+        )
+        from uaclient.entitlements import ENTITLEMENT_CLASS_BY_NAME
+
+        contract_information = get_contract_information(self, token)
+
+        contract_info = contract_information.get("contractInfo", {})
+        account_info = contract_information.get("accountInfo", {})
+
+        response = {
+            "version": version.get_version(features=self.features),
+            "contract": {
+                "id": contract_info.get("id", ""),
+                "name": contract_info.get("name", ""),
+                "created_at": contract_info.get("createdAt", ""),
+                "products": contract_info.get("products", []),
+            },
+            "account": {
+                "name": account_info.get("name", ""),
+                "id": account_info.get("id"),
+                "created_at": account_info.get("createdAt", ""),
+                "external_account_ids": account_info.get(
+                    "externalAccountIDs", []
+                ),
+            },
+            "services": [],
+            "execution_status": DEFAULT_STATUS.get("execution_status", ""),
+            "simulated": True,
+        }
+
+        if contract_info.get("effectiveTo"):
+            response["expires"] = contract_info.get("effectiveTo")
+        if contract_info.get("effectiveFrom"):
+            response["effective"] = contract_info.get("effectiveFrom")
+
+        resources = get_available_resources(self)
```

Comment:
Isn't the information we want already present in the `contract_information` variable ?

Reply:
Yes, you are right, we should be checking for that in the cache before making the call to the server - will fix.

Reply:
added a check for the status cache before making this call